src/Symfony/Component/AssetMapper/composer.json

```diff
@@ -17,6 +17,7 @@
     ],
     "require": {
         "php": ">=8.1",
+        "composer/semver": "^3.0",
```

Comment:
Be careful about that. The constraints are not 100% compatible between the npm and composer ecosystems.

Reply:
Hmm, is there an alternative? Checking into this quickly, differences I found:

* npm has a `1.0.0 - 2.0.0` syntax.
* npm seems to have a `5.4.x` syntax which means `5.4.*`.

I think this feature is pretty important, without it, it gets easy for your dependencies to "drift" out of sync (e.g. you install `boostrap` today and we also give you `@popper/core`, then you upgrade bootstrap but forget to upgrade popper and have no warning). So perhaps, IF we run into problems with odd, npm-specific dependency things we can handle those manually? Or maybe we can handle the above 2 items right now manually.

Reply:
IIRC, the `~` operator does not work the same either

Reply:
I've added a simple conversion system using all of the differences I could find. There are some edge-case things that can't be converted - e.g. `stable` or `https://example.com/file.js`, but I think that's ok. This is just a system to warn users of potential problems (like the peer dependency warning in npm). And if we're missing some real-world cases, we can find out about them and patch them. The "conversion" code is fairly minimal, as the differences don't appear to be too many. 

Thanks for catching this!